zenpacklib.py

```diff
@@ -2448,8 +2448,10 @@ def datapoint_method(self, default=spec.datapoint_default, cached=spec.datapoint
                     if cached:
                         r = self.cacheRRDValue(datapoint, default=default)
                     else:
-                        r = self.getRRDValue(datapoint, default=default)
-
+                        if is_pre_zenoss_5():
```

Comment:
It seems unnecessarily expensive to make this check every time getRRDValue needs to be called when its result is always the same thing. I'd recommend setting a zenpacklib module global that figures this out once so we're only checking its value here.

Something like this (use feature checking, not explicit version checking) in the zenpacklib module scope.

```python
try:
    from Products.Zuul.facades import metricfacade  # noqa: imported only to test that it can be imported
except ImportError:
    HAS_METRICFACADE = False
else:
    HAS_METRICFACADE = True
```

Then switch this section to..

```python
if HAS_METRICFACADE:
    r = self.getRRDValue(datapoint)
else:
    r = self.getRRDValue(datapoint, start=time.time()-1800)
```